src/signing/mod.rs

```diff
@@ -97,9 +97,23 @@ impl Deref for SignerHandle {
 #[async_trait::async_trait]
 pub trait Signer: Send + Sync {
     /// Get the ledger status.
-    async fn get_ledger_status(&self, is_simulator: bool) -> LedgerStatus;
+    ///
+    /// This is only meaningful for the Ledger hardware; other signers don't implement this.
+    async fn get_ledger_status(&self, _: bool) -> LedgerStatus {
+        LedgerStatus {
+            app: None,
+            connected: false,
+            locked: false,
+        }
+    }
+
     /// Initialises a mnemonic.
-    async fn store_mnemonic(&mut self, storage_path: &Path, mnemonic: String) -> crate::Result<()>;
+    ///
+    /// This is only meaningful for the Stronghold signer; other signers don't implement this.
+    async fn store_mnemonic(&mut self, _: &Path, _: String) -> crate::Result<()> {
+        Ok(())
```

Comment:
Should we rather return an error, because a user could otherwise expect that it gets stored _somewhere_?

Reply:
Which error can we return, or we crate a new error specifically for this case?

Reply:
We would need a new one